packages/cohort_banking/src/callbacks/state_provider.rs

```diff
@@ -118,18 +131,35 @@ impl StateProviderImpl {
                     version: tuple.0.version,
                 })
                 .collect(),
-            abort_reason: None,
         })
     }
-}
 
-#[async_trait]
-impl ItemStateProvider for StateProviderImpl {
-    async fn get_state(&self) -> Result<CapturedState, String> {
-        if self.single_query_strategy {
-            self.get_state_using_one_query().await
+    pub async fn get_certification_candidate(&self, request: CertificationRequest) -> Result<CertificationCandidateCallbackResponse, String> {
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        // This example doesn't handle `Cancelled` scenario.
+        // If user cancellation is needed, add additional logic in this fn to return `Cancelled` instead of `Proceed` in the result.
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+        let mut accounts: Vec<&(dyn ToSql + Sync)> = vec![];
+        request.candidate.readset.iter().for_each(|x| accounts.push(x));
+
+        let state = if self.single_query_strategy {
+            self.get_state_using_one_query(&accounts).await
         } else {
-            self.get_state_using_two_queries().await
-        }
+            self.get_state_using_two_queries(&accounts).await
```

Comment:
I don't like this part where signature is dynamic accounts array. In this app we will have 2 accounts as fixed structure. It is also used like that in SQL query. Masking it into param which accepts a dynamic array looks confusing.

Reply:
agree, I was experimenting something and just left it there, I have modified this now.